### PR TITLE
Option to invert the returned value for binary_sensors

### PR DESCRIPTION
**Pull request by ydubreuil**
This adds an option to invert the returned value of a binary_sensors. I needed this to report the value of a dry contact checking if my garage door is closed or not.

Thanks for this integration, it works well for me, just needed this little change :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -206,6 +206,10 @@ default_brightness:
   description: default brightness for xpwm, xpwm_rgb and xpwm_rgbw only for turn on command (must be between 1 and 255)
   required: false
   type: int
+invert:
+  description: invert the value returned for binary_sensors (on become off and vice versa)
+  required: false
+  type: bool
```
Comment (Aohzan) [inline README.md:212]: add a default to False, to assure retrocompat

File: custom_components/ipx800v4/__init__.py [hunk 2]
```diff
@@ -81,6 +82,7 @@
         vol.Required(CONF_TYPE): cv.string,
         vol.Optional(CONF_ID): cv.positive_int,
         vol.Optional(CONF_IDS): cv.ensure_list,
+        vol.Optional(CONF_INVERT): cv.boolean,
```
Comment (Aohzan) [inline custom_components/ipx800v4/__init__.py:85]:
```suggestion
        vol.Optional(CONF_INVERT, default=False): cv.boolean,
```

File: custom_components/ipx800v4/binary_sensor.py [hunk 3]
```diff
@@ -36,26 +40,50 @@ async def async_setup_entry(
 
     for device in devices:
         if device.get(CONF_TYPE) == TYPE_VIRTUALOUT:
-            entities.append(VirtualOutBinarySensor(device, controller, coordinator))
+            on_value = 0 if device.get(CONF_INVERT) else 1
+            entities.append(VirtualOutBinarySensor(device, controller, coordinator, on_value))
```
Comment (Aohzan) [inline custom_components/ipx800v4/binary_sensor.py:44]: you don't have to handle it here

File: custom_components/ipx800v4/const.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 CONF_DEFAULT_BRIGHTNESS = "default_brightness"
 CONF_ID = "id"
 CONF_IDS = "ids"
+CONF_INVERT = "invert"
```
Comment (Aohzan) [inline custom_components/ipx800v4/const.py:20]: I think `CONF_INVERTED_VALUE` is more explicit, no ?

File: custom_components/ipx800v4/binary_sensor.py [hunk 5]
```diff
@@ -36,26 +40,50 @@ async def async_setup_entry(
...
+        self,
+        device_config: dict,
+        ipx: IPX800,
+        coordinator: DataUpdateCoordinator,
+        on_value: int
+    ) -> None:
+        """Initialize the VirtualOutBinarySensor."""
+        super().__init__(device_config, ipx, coordinator)
+        self.on_value = on_value
+
```
Comment (Aohzan) [inline custom_components/ipx800v4/binary_sensor.py:65]: as we now set the Inverted value to False for all config entities, you can defined `self._inverted_value = device_config[CONF_INVERT]` directly in the class IpxEntity

File: custom_components/ipx800v4/binary_sensor.py [hunk 6]
```diff
@@ -36,26 +40,50 @@ async def async_setup_entry(
...
     @property
     def is_on(self) -> bool:
         """Return the state."""
-        return self.coordinator.data[f"VO{self._id}"] == 1
+        return self.coordinator.data[f"VO{self._id}"] == self.on_value
```
Comment (Aohzan) [inline custom_components/ipx800v4/binary_sensor.py:69]: and after that:
```suggestion
        return self.coordinator.data[f"VO{self._id}"] == 1 if not self._inverted_value else 0
```

**Review by Aohzan — CHANGES_REQUESTED** [review]
Thanks for the contribution

Can you fill the CHANGELOG too please

File: README.md (hunk 1 above)
Comment (ydubreuil) [reply README.md:212]: Not setting the value in `configuration.yaml` is equivalent to setting it to false. I'll try to find how to add a default

File: custom_components/ipx800v4/__init__.py (hunk 2 above)
Comment (ydubreuil) [reply custom_components/ipx800v4/__init__.py:85]: ah nice, never used that API before

File: custom_components/ipx800v4/binary_sensor.py (hunk 3 above)
Comment (ydubreuil) [reply custom_components/ipx800v4/binary_sensor.py:44]: I initially implemented that below but my python fu is so bad that it wasn't working, I'll try to move that down again

File: custom_components/ipx800v4/const.py (hunk 4 above)
Comment (ydubreuil) [reply custom_components/ipx800v4/const.py:20]: Good idea, I'll rename it. I'd keep the infinitive form of the verb though, it's an action.

File: custom_components/ipx800v4/binary_sensor.py (hunk 5 above)
Comment (ydubreuil) [reply custom_components/ipx800v4/binary_sensor.py:65]: yeah, make sense. I guess that's the reason why my first attempt wasn't working

**Comment by ydubreuil** [comment]
Thanks for the review, I addressed your comments and added an entry in the CHANGELOG.

File: README.md (hunk 1 above)
Comment (Aohzan) [reply README.md:212]:
```suggestion
  type: bool
  default: false
```

File: custom_components/ipx800v4/binary_sensor.py (hunk 6 above)
Comment (ydubreuil) [reply custom_components/ipx800v4/binary_sensor.py]: this introduced a bug that I fixed in https://github.com/Aohzan/ipx800/pull/48/commits/89fd13fa5e1b5e4fb63b81cdc4aded4dda0837e3

**Comment by Aohzan** [comment]
I pushed a commit to enhance readme and set invert value in parent class

**Comment by Aohzan** [comment]
can you test with the last modifications and tell me if everything is working ? (I don't have a V4 now to test)

**Comment by ydubreuil** [comment]
Just tested your changes, they work fine.

**Comment by Aohzan** [comment]
thanks @ydubreuil

**Pull request merged**